#184948256 Update password every x amount of time

Rwibutso opened the pull request:
#### What does this PR do?
This PR  prompt an interface to user to change password for x amount of timet
#### Description of Task to be completed?
-  Create a UI for the task
- Code the functionality
- Integrate with the backend
- Test the application
#### How should this be manually tested?
- clone this repo `git clone https://github.com/atlp-rwanda/e-comm-team-amigos-fn`
- switch your working directory `cd e-comm-team-amigos-fn`
- create a `.env` and update it with `.env-example`
- install all dependencies by running `npm install`
- in your terminal, run `npm start`
#### Test the endpoints:
-  `/update-password`
#### Any background context you want to provide?
- Node JS installed
- Node server running
#### What are the relevant pivotal tracker stories?
Story ID: [#184948256](https://www.pivotaltracker.com/story/show/184948256)
####Screenshots
![Screenshot from 2023-04-28 12-14-22](https://user-images.githubusercontent.com/54440360/235121876-0e2e4c60-d8d1-40f7-b27f-fcacc13d2765.png)
![Screenshot from 2023-04-28 12-14-34](https://user-images.githubusercontent.com/54440360/235121904-a19b4229-3e3b-4ef2-96a6-8e33d6a2de61.png)


![Screenshot from 2023-05-02 14-19-09](https://user-images.githubusercontent.com/54440360/235664477-322993a1-3661-4ae5-86f4-d514dbbf941e.png)

Rwibutso committed:
feat(update-password): change password every x amount of time

- prompt an interface to user to change their password for x amount of time

[Delivers #184948256]

Emile-SHUMBUSHO commented:
This PR has a conflict to resolve

Emile-SHUMBUSHO commented on src/assets/css/changePassword.css:
```diff
@@ -0,0 +1,147 @@
+        margin-top: 0px;
+        font-size: 15px;
+    }
+} 
```
Avoid an empty line from your code

Emile-SHUMBUSHO commented on src/components/changePasswordForm.jsx:
```diff
@@ -0,0 +1,109 @@
+    );
+  };
+  
+  export default ChangePasswordPage;
```
Avoid an empty line from your code

Emile-SHUMBUSHO commented on src/views/changePasswordPage.jsx:
```diff
@@ -0,0 +1,8 @@
+			<ChangePasswordForm />
+		</>
+	);
+}
```
Avoid an empty line from your code

Emile-SHUMBUSHO commented on src/components/changePasswordForm.jsx:
```diff
@@ -0,0 +1,109 @@
+      },
+      validationSchema: validationSchema,
+      onSubmit:async (values) => {
+        try{
+            const res = await axios.put("https://e-comm-team-amigos-bn-project.onrender.com/user/createPassword", {email:values.email,oldpassword:values.oldPassword,newpassword:values.newPassword});
+            // localStorage.setItem("token",res.data.token);
+            console.log("success",res.data);
+            updateErrorDisplay.innerHTML = "Password updated successfuly!";
+            updateErrorDisplay.style.color = "#096F3E";
+            useEffect(()=>{
+              if(updatingPasswordSuccess)
+              toast(updatingPasswordSuccess)
+              if(updatingPasswordError)
+              toast(updatingPasswordError)
+              if(updatingPasswordSuccess === 'password updated successfully'){
+                  setTimeout(()=>{
+                      navigation("https://e-comm-team-amigos-bn-project.onrender.com/user/login")
+                  },5000)
+              }
+          },[updatingPasswordSuccess, updatingPasswordError])
+            console.log(res);
+        }catch(error){
+            if (error.response.status == 401) {
+              updateErrorDisplay.innerHTML = "Incorect email or old password!";
+              updateErrorDisplay.style.color = "red";
+            }else if (error.response.status == 404) {
+                updateErrorDisplay.innerHTML = "Email not found!";
+                updateErrorDisplay.style.color = "red";
+            }else {
+                console.log(error);
+            }
+        }
+    },
+  });
```
Great Job for making integration with API but I'm suggesting to separate Formik validation and integration functionality. Further more, take advantage of using redux to manage states  in your component.

AbdulKhaliq59 reviewed (approved):
Rebase Unless it LGTM

Rwibutso committed:
feat(update-password): change password every x amount of time

- prompt an interface to user to change their password for x amount of time

[Delivers #184948256]

Habinezajanvier commented on src/views/updatePassword/updatePassword.style.scss:
```diff
@@ -0,0 +1,148 @@
+//     h3{
+//         font-size: 16px;
+//     } 
+//     .inputField{
+//         height: 40px;
+//         width: 80%;
+//         outline: none;
+//         margin: 8px;
+//     }
+//     .form{
+//         margin-top: 0px;
+//     }
+//     .submitBtn{
+//         height: 40px;
+//         width: 80%;
+//         padding: 7px;
+//         margin-top: 0px;
+//         font-size: 15px;
+//     }
+// } 
```
Why is those commented codes, if they are not needed remove them at all

Habinezajanvier reviewed (approved):
👍
